crates/matrix-sdk-sqlite/src/crypto_store.rs

```diff
@@ -270,6 +277,11 @@ async fn run_migrations(conn: &SqliteConn) -> Result<(), CryptoStoreError> {
     }
 
     if version < 1 {
+        // First turn on WAL mode, this can't be done in the transaction, it fails with
+        // the error message: "cannot change into wal mode from within a transaction".
+        conn.execute_batch("PRAGMA journal_mode = wal;")
```

Comment:
Undo it? Why? We want to have it enabled forever.

Reply:
Oh, but as-is we only enable it for the connection that is used for migrations, and only if the initialization migration is run.

Reply:
That's fine, from the SQLite docs:

> Unlike the other journaling modes, [PRAGMA journal_mode=WAL](https://www.sqlite.org/pragma.html#pragma_journal_mode) is persistent. If a process sets WAL mode, then closes and reopens the database, the database will come back in WAL mode. In contrast, if a process sets (for example) PRAGMA journal_mode=TRUNCATE and then closes and reopens the database will come back up in the default rollback mode of DELETE rather than the previous TRUNCATE setting.

And:

> The WAL journal mode will be set on all connections to the same database file if it is set on any one connection. 

More info in section 3.3 here: https://www.sqlite.org/wal.html